bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/gems/CandidateGuideStarsTableModel.java

```diff
@@ -137,16 +140,21 @@ public boolean isCellEditable(int row, int column) {
     }
 
     @SuppressWarnings("unchecked")
-    public TableQueryResult getTableQueryResult() {
-        String raPosition = String.valueOf(_columnNames.indexOf(RA_TITLE));
-        String decPosition = String.valueOf(_columnNames.indexOf(DEC_TITLE));
-        Properties props = new Properties();
-        props.setProperty(SkycatConfigFile.RA_COL, raPosition);
-        props.setProperty(SkycatConfigFile.DEC_COL, decPosition);
-        props.setProperty(SkycatConfigFile.SERV_TYPE, "catalog");
-        props.setProperty(SkycatConfigFile.LONG_NAME, "ucac4");
-        SkycatConfigEntry entry = new SkycatConfigEntry(props);
-        SkycatTable skycatTable = new SkycatTable(entry, getDataVector(), getFields()) {
+    TableQueryResult getTableQueryResult() {
+        final String raPosition = String.valueOf(_columnNames.indexOf(RA_TITLE));
+        final String decPosition = String.valueOf(_columnNames.indexOf(DEC_TITLE));
+        final Properties props = new Properties();
+        props.setProperty(RA_COL, raPosition);
+        props.setProperty(DEC_COL, decPosition);
+        props.setProperty(SERV_TYPE, "catalog");
+        props.setProperty(LONG_NAME, "ucac4");
+        final SkycatConfigEntry entry = new SkycatConfigEntry(props);
+        return new SkycatTable(entry, CandidateGuideStarsTableModel.this.getDataVector(), getFields()) {
+            @Override
+            public Option<SiderealTarget> getSiderealTarget(int i) {
```

Comment:
This is the key change for the PR. This method is called from the TPE to find out if a given point (translated into a query result row id) correspond to a target. By default this returns `None` making the selection fail

Most of the other changes come down from this change. In particular the order of items in `_model` is critical since we are looking up by row number and getting the order right trickles downstream